### PR TITLE
refactor: reduce direct access to OutBuffer fields

### PR DESCRIPTION
**Pull request by WalterBright**
Because accessing fields directly is bad.

### CODE REVIEW

**Comment by WalterBright** [comment]
What line in the code are you referring to?

File: src/doc.d [hunk 1]
```diff
@@ -2562,15 +2565,15 @@ extern (C++) void highlightCode(Scope* sc, Dsymbols* a, OutBuffer* buf, size_t o
                 }
                 parametersBuf.writeByte(')');
 
-                const(char*) templateParams = parametersBuf.extractString();
-                size_t templateParamsLen = strlen(templateParams);
+                const templateParams = parametersBuf.peekString();
+                const templateParamsLen = strlen(templateParams);
```
Comment (yebblies) [inline src/doc.d]: It was this one.

Comment (WalterBright) [inline src/doc.d]: ok

File: src/doc.d [hunk 2]
```diff
@@ -2563,7 +2566,7 @@ extern (C++) void highlightCode(Scope* sc, Dsymbols* a, OutBuffer* buf, size_t o
                 parametersBuf.writeByte(')');
 
                 const templateParams = parametersBuf.peekString();
-                const templateParamsLen = strlen(templateParams);
+                const templateParamsLen = parametersBuf.peekSlice().length;
```
Comment (yebblies) [inline src/doc.d]: Can't we have a nice way to access length?

Comment (WalterBright) [inline src/doc.d]: I thought about it, but there weren't enough use cases. Inlining peekSlice() should take care of any efficiency concerns.

Comment (yebblies) [inline src/doc.d]: Hmm, `peekSlice` should probably be `opSlice`...

**Comment by yebblies** [comment]
Auto-merge toggled on

**Pull request merged**